conn.go

```diff
@@ -454,6 +433,10 @@ func (s *startupCoordinator) options(ctx context.Context) error {
 	// Keep raw supported multimap for debug purposes
 	s.conn.supported = v.supported
 	s.conn.scyllaSupported = parseSupported(s.conn.supported)
+	s.conn.host.mu.Lock()
+	s.conn.host.scyllaShardAwarePort = s.conn.scyllaSupported.shardAwarePort
+	s.conn.host.scyllaShardAwarePortTLS = s.conn.scyllaSupported.shardAwarePortSSL
+	s.conn.host.mu.Unlock()
```

Comment:
Refactor to method in host. Taking other object's mutex from outside is asking for trouble.

Reply:
Good point. Moved. Thanks!